src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/finance/TransactionSynchronisationIntTest.kt

```diff
@@ -0,0 +1,829 @@
+import com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor
+import com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo
+import com.github.tomakehurst.wiremock.client.WireMock.urlPathMatching
+import org.assertj.core.api.Assertions.assertThat
+import org.awaitility.kotlin.await
+import org.awaitility.kotlin.matches
+import org.awaitility.kotlin.untilAsserted
+import org.awaitility.kotlin.untilCallTo
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.mockito.ArgumentMatchers.eq
+import org.mockito.kotlin.atLeast
+import org.mockito.kotlin.check
+import org.mockito.kotlin.isNull
+import org.mockito.kotlin.verify
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR
+import org.springframework.http.HttpStatus.NOT_FOUND
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.finance.FinanceApiExtension.Companion.financeApi
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.finance.model.SyncTransactionReceipt
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.integration.SqsIntegrationTestBase
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.integration.sendMessage
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomismappings.model.TransactionMappingDto
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomisprisoner.model.GeneralLedgerTransactionDto
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nomisprisoner.model.OffenderTransactionDto
+import uk.gov.justice.digital.hmpps.prisonerfromnomismigration.wiremock.withRequestBodyJsonPath
+import uk.gov.justice.hmpps.sqs.countAllMessagesOnQueue
+import java.math.BigDecimal
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.*
+import java.util.AbstractMap.SimpleEntry
+
+private const val BOOKING_ID = 1234L
+private const val NOMIS_TRANSACTION_ID = 2345678L
+private const val OFFENDER_ID = 101L
+private const val OFFENDER_ID_DISPLAY = "A3864DZ"
+private const val DPS_TRANSACTION_ID = "a04f7a8d-61aa-400c-9395-000011112222"
+private const val MESSAGE_ID = "abcdef01-0000-1111-2222-000011112222"
+
+class TransactionSynchronisationIntTest : SqsIntegrationTestBase() {
+
+  private val dpsTransactionUuid = UUID.fromString(DPS_TRANSACTION_ID)
+  private val messageUuid = UUID.fromString(MESSAGE_ID)
+
+  @Autowired
+  private lateinit var financeNomisApiMockServer: FinanceNomisApiMockServer
+
+  @Autowired
+  private lateinit var financeMappingApiMockServer: FinanceMappingApiMockServer
+
+  @Nested
+  @DisplayName("OFFENDER_TRANSACTIONS-INSERTED")
+  inner class TransactionInserted {
+
+    @Nested
+    @DisplayName("When transaction was created in DPS")
+    inner class DPSCreated {
+
+      @BeforeEach
+      fun setUp() {
+        awsSqsFinanceOffenderEventsClient.sendMessage(
```

Comment:
You can just do financeOffenderEventsQueue.sendMessage

Reply:
ok will change